core/SymbolRef.h

```diff
@@ -29,19 +33,54 @@ class SymbolData : private DebugOnlyCheck<SymbolDataDebugCheck> {
 };
 CheckSize(SymbolData, 8, 8);
 
+class Symbol;
+
 class SymbolRef final {
     friend class GlobalState;
     friend class Symbol;
 
+    u4 _id;
+
 public:
-    SymbolRef(GlobalState const *from, u4 _id);
-    SymbolRef(const GlobalState &from, u4 _id);
+    // TODO: Enforce that this is only 3 bits?
+    enum class Kind : u1 {
+        ClassOrModule = 0,
+        Method = 1,
+        Field = 2,
+        TypeArgument = 3,
```

Comment:
not for this pr, but we should probably move type arguments out of symbol table(as we did with method args, block args and blocks themselves)